### PR TITLE
test(application-system): Add small acceptance test for application-system

### LINKED ISSUE #7664: Test Case: Starting a new application should create a new application under “your applications”

### PR DESCRIPTION
**Pull request by kristofer-lp8f**
Closes #7664

## What
This PR adds more tests to prevent accidental breakages.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by kristofer-lp8f** [comment]
Rebase onto newest main to prevent CI  change detection errors

File: apps/system-e2e/src/integration/application-system/p-sign.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    cy.visit('/umsoknir/p-merki')
+
+    // First page load takes a long time
+    cy.contains('Gagnaöflun', { timeout: 10000 })
+    cy.pathUuid().should('match', uuidRegex.v4)
+
+    // Data collection
+    cy.get('input[name="approveExternalData"]').click()
+    cy.get('button[type="submit"]').click()
+    cy.wait(5000)
```
Comment (busla) [inline apps/system-e2e/src/integration/application-system/p-sign.spec.ts:29]: `click()` can return a callback when the response is return so explicit `wait()` is probably not needed

**Review by busla — CHANGES_REQUESTED** [review]
The integration should probably be nested under `smoke/` or `acceptance/`

File: package.json [hunk 2]
```diff
@@ -381,7 +381,7 @@
     "cross-spawn": "7.0.3",
     "csstype": "2.6.13",
     "csv-parse": "4.16.0",
-    "cypress": "10.2.0",
+    "cypress": "10.3.1",
```
Comment (busla) [inline package.json:384]: Make sure to also bump the Cypress version in the Dockerfile https://github.com/island-is/island.is/blob/main/scripts/ci/Dockerfile#L167

**Review by busla — CHANGES_REQUESTED** [review]

**Review by eirikurn — APPROVED** [review]

File: apps/system-e2e/src/integration/application-system/smoke/p-sign.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    cy.visit('/umsoknir/p-merki')
+
+    // First page load takes a long time
+    cy.contains('Gagnaöflun', { timeout: Timeout.long })
+    cy.pathUuid().should('match', uuidRegex.v4)
+
+    // Data collection
+    cy.get('input[name="approveExternalData"]').click()
+    cy.get('button[type="submit"]').click()
+    cy.get('[data-testid="alertMessage"]').should('not.exist')
```
Comment (busla) [inline apps/system-e2e/src/integration/application-system/smoke/p-sign.spec.ts:26]:
```suggestion
    cy.get('button[type="submit"]').click().get('[data-testid="alertMessage"]').should('not.exist')
```

File: apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts [hunk 4]
```diff
@@ -1,17 +1,19 @@
+import { getFakeUser } from '../../../support/utils'
 import fakeUsers from '../../../fixtures/service-portal/users.json'
 
 describe('Service Portal', () => {
+  const userName = 'María Sól ÞÍ Torp'
```
Comment (busla) [inline apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts]:
```suggestion
```

File: apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
+  const testUser = getFakeUser(fakeUsers, userName)
   beforeEach(() => {
     cy.idsLogin({
-      phoneNumber: fakeUsers[0].phoneNumber,
-      baseUrl: Cypress.config('baseUrl'),
+      phoneNumber: testUser,
```
Comment (busla) [inline apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts]:
```suggestion
      phoneNumber: testUser.phoneNumber,
```

File: apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
       urlPath: '/minarsidur/',
     })
   })
 
-  it(`should have user ${fakeUsers[0].name} logged in`, () => {
+  it(`should have ${userName} logged in`, () => {
```
Comment (busla) [inline apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts]:
```suggestion
  it(`should have ${fakeUser.name} logged in`, () => {
```

File: apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts [hunk 7]
```diff
@@ -1,17 +1,19 @@
...
     cy.visit('/minarsidur/')
-    cy.contains(fakeUsers[0].name)
+    cy.contains(userName)
```
Comment (busla) [inline apps/system-e2e/src/integration/service-portal/smoke/inbox.spec.ts]:
```suggestion
    cy.contains(fakeUser.name)
```

**Review by busla — CHANGES_REQUESTED** [review]
LGTM

minor adjustments

File: apps/system-e2e/src/lib/types.ts [hunk 8]
```diff
@@ -35,6 +35,7 @@ enum BaseAuthority {
   ads = 'loftbru.dev01.devland.is',
   prod = 'island.is',
   local = 'localhost:4200',
+  applicationSystem = 'localhost:4242',
```
Comment (pshomov) [inline apps/system-e2e/src/lib/types.ts:38]: I have added in the docs that we can (and should) run application system web on port 4200. Alternatively we need a better configuration support for local webs running on different ports.

**Review by pshomov — APPROVED** [review]
LGTM

File: apps/system-e2e/src/lib/types.ts (hunk 8 above)
Comment (kristofer-lp8f) [reply apps/system-e2e/src/lib/types.ts:38]: Should I revert this?

File: apps/system-e2e/src/support/commands.ts [hunk 9]
```diff
@@ -75,7 +79,16 @@ Cypress.Commands.add(
...
   'idsLogin',
-  ({ phoneNumber, authUrl, baseUrl, urlPath = '/' }) => {
+  ({ phoneNumber, authUrl, baseUrl, urlPath = '/', setBaseUrl = true }) => {
+    if (setBaseUrl) {
+      // FIXME: https://github.com/island-is/island.is/issues/8261
+      if (Cypress.env().testEnvironment == 'local') {
+        const newBaseUrl = getEnvironmentBaseUrl(baseUrl)
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        Cypress.config('baseUrl', newBaseUrl)
```
Comment (busla) [inline apps/system-e2e/src/support/commands.ts:89]: are you sure you can change the baseUrl in a `beforeEach` call?
 
 I think this needs to happen in cypress config.

**Review by busla — APPROVED** [review]
LGTM

one question about baseUrl

File: apps/system-e2e/src/support/commands.ts (hunk 9 above)
Comment (kristofer-lp8f) [reply apps/system-e2e/src/support/commands.ts:89]: Nope. This is leftovers from when I changed the port to 4242. The non-4200 port issue should be fixed seperately.

Comment (kristofer-lp8f) [reply apps/system-e2e/src/support/commands.ts]: Removed.

**Pull request merged**